WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -263,5 +263,20 @@ public static void SortByAmount(this TxOutList list)
 		{
 			list.Sort((x, y) => x.Value.CompareTo(y.Value));
 		}
+
+		public static void SortByAmount(this TxInList list, List<Coin> coins)
+		{
+			Predicate<TxIn> txInFinder;
+			var map = new Dictionary<TxIn, Coin>();
+			foreach (var coin in coins)
+			{
+				txInFinder = (TxIn txIn) =>
```

Comment:
I don't get why we need the finder here. I would suggest something simpler.

**Pseudocode:**
```c#
public static void SortByAmount(this TxInList list, List<Coin> coins)
{
	var map = new Dictionary<TxIn, Coin>();
	foreach (var coin in coins)
	{
		var txin = list.Single(x=>x.PrevOut == coin.Outpoint);
		map.Add(txin, coin);
	}
	list.Sort((x, y) => map[x].Amount.CompareTo(map[y].Amount));
}
```

Reply:
I didn't know you could use a Predicate with `.Single()`, thanks